lib/utils.pm

```diff
@@ -867,10 +867,15 @@ sub get_x11_console_tty {
       && !check_var('VIRSH_VMM_FAMILY', 'hyperv')
       && !check_var('VIRSH_VMM_TYPE',   'linux')
       && !get_var('VERSION_LAYERED');
-    # starting with Tumbleweed build 20190614, we ship gdm 3.32 and gnome
-    # desktop runs on tty2 for both auto-login cases and non auto-login cases
-    my $gdm_3_32 = is_tumbleweed() && get_var('BUILD') ge "20190614";
-    return (check_var('DESKTOP', 'gnome') && (get_var('NOAUTOLOGIN') || $gdm_3_32) && $new_gdm) ? 2 : 7;
+    # $newer_gdm means GDM version >= 3.32, which will start gnome desktop
+    # on tty2 including auto-login cases. Also exclude cases which boot from
+    # older versions in HDD
+    my $newer_gdm
```

Comment:
I think "gdm_3_32" is better than "newer_gdm" as we already have "new_gdm" but it's ok either way